src/engine/guidance/post_processing.cpp

```diff
@@ -498,6 +498,10 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
     BOOST_ASSERT(one_back_index < steps.size());
     const auto &current_step = steps[step_index];
     const auto &one_back_step = steps[one_back_index];
+    // Don't collapse roundabouts
+    if (entersRoundabout(current_step.maneuver.instruction) ||
+        entersRoundabout(one_back_step.maneuver.instruction))
+        return;
```

Comment:
No test so far checked for this situation. So our tests couldn't catch it. It's likely that the regression came in with the changes of turn angles, since I modified postprocessing a bit as well.

What are you calling a `multi-hop roundabout instruction`? During turn processing, we only have `enter roundabout` left. All other roundabout instructions should have been removed by now.